RAT/project.py

```diff
@@ -524,11 +525,12 @@ def wrapped_func(*args, **kwargs):
             try:
                 return_value = func(*args, **kwargs)
                 Project.model_validate(self)
-            except ValidationError as e:
+            except ValidationError as exc:
                 setattr(class_list, 'data', previous_state)
-                error_string = formatted_pydantic_error(e)
-                # Use ANSI escape sequences to print error text in red
-                print('\033[31m' + error_string + '\033[0m')
+                traceback_string = formatted_traceback()
```

Comment:
I think we still need to [propagate the exception](https://stackoverflow.com/questions/73719034/how-to-properly-propagate-error-messages-using-python) with the new error message, otherwise if logging is switched off or an app has no console (typical GUI app), there will be no way for upstream packages to know there is a problem. I am not sure if there is a specific reason the exception is not re-raised

maybe
```
        traceback_string = formatted_traceback()
        error_string = formatted_pydantic_error(exc, custom_msgs)
        raise ValueError(traceback_string + error_string) from None
```

or

```
        error_string = formatted_pydantic_error(exc, custom_msgs)
        raise ValueError(error_string) from exc
```

 PS: I used ValueError cause I could not create ValidationError. This way I think the logging is not need

Reply:
Pydantic recommends this way of handling ValidatioErrors: https://docs.pydantic.dev/latest/errors/errors/ I'm inclined to agree that raising the error is a better cause of action in general, but I don't know how to do it with the error message we want - I'll have a look and see what I can find though.

Reply:
i.e. - it's dead easy to raise the exception with the standard error message, but that it what we want to change.

Reply:
I don't mind if the exception is propagated as is and the upstream modifies the message but their suggestions are for end users only and will not help if there is no console which will be the case with RASCAL. If the best solution is to reraise the exception without modification then you have to modify this code so it doesn't print the same thing twice

Reply:
You could also make a custom RAT validation error and copy the traceback